lib/gst/player/gstplayer.c

```diff
@@ -112,6 +122,9 @@ struct _GstPlayerPrivate
   GstPlayerState app_state;
   gint buffering;
 
+  GstPlayerMediaInfo  *media_info;  /* current media info */
+  GstPlayerMediaInfo  *media_info_copy; /* last emited media info */
```

Comment:
Why not:
- whenever something changes, create a new one or copy. Then modify that one
- if had an old one, unref that one
- store the new one
- emit the signal with the new one

It's not needed to keep the old one around until you emitted the signal for the new one